### DISCUSSION TITLE
Registering 2D RGB images

### DISCUSSION THREAD

**Discussion by kretep** [SuperElastix/elastix#1190] [Q&A] [answered]
Hi, I'd like to register two 2D RGB images (16-bit unsigned int, in TIFF format). This doesn't seem to work out of the box. I tried extracting the red channel and registering those, but the 2D 16-bit pixel type doesn't seem to be available. I also tried converting the RGB image to a 3-layered TIFF image and that seems to work in that it performs the registration, but also in the through-stack dimension. Is it possible to constrain the transform to 2D? Or is there a different approach that I might be missing?

**Comment by N-Dekker** [comment]
Hi Peter! Thanks for your question! Do you use the elastix executable or the library interface? Or one of the Python wrappers, SimpleElastix, SimpleITK, or ITKElastix?

**Reply by kretep** [reply]
Hi Niels, at the moment I'm trying the executable. (I also considered the wrappers, but for various reasons they didn't seem to work out-of-the-box on my Apple Silicon and I prefer not having to build the binaries myself.)

**Reply by N-Dekker** [reply, marked as answer]
OK, so then can you possibly try "float" as internal pixel type, for your 2D 16-bit unsigned images? In your parameter txt files, as follows:

```
(FixedInternalImagePixelType "float")
(MovingInternalImagePixelType "float")
```
Internally, elastix will then convert your pixel data to 32-bit float, but it should still work fine. At least, it should!

**Reply by kretep** [reply]
Thanks, that seems to work in that the registration runs, but what do I then use as the ResultImagePixelType? I tried various combinations using only one-channel images for now. 

When I use "unsigned short", the image seems completely black. When applying transformix on the same image, the image also seems black, but I can retrieve (some) information by dramatically increasing the contrast.

When I do the registration on 8-bit images and use "unsigned char", the result does show the image. But my goal is to preserve all the information that is in the 16-bit images... any ideas?

**Reply by N-Dekker** [reply]
Maybe the final interpolation used when generating the result caused the output image to look black. Which value do you have for `ResampleInterpolator`? I'm not sure, but maybe you can try nearest neighbor:

    (ResampleInterpolator "FinalNearestNeighborInterpolator")

You can probably just run transformix, it should not be necessary to rerun the registration.

Please let us know if that helps!

**Reply by kretep** [reply]
I had another go at debugging it and it turned out that when I extracted the color channels, it converted to 8-bit images by default instead of 16-bit. So after making sure I got the 16-bit inputs, it now works using float as the internal pixel type.
